core/app/models/spree/order/address_book.rb

```diff
@@ -30,7 +30,7 @@ def bill_address_id=(id)
 
       def bill_address_attributes=(attributes)
         self.bill_address = update_or_create_address(attributes)
-        user.bill_address = bill_address if user && user.bill_address.nil?
+        user.update(bill_address: bill_address) if user && user.bill_address.nil?
```

Comment:
A note on this change: simply assigning the `user.bill_address` will not persist the `bill_address` on the user. However, the `user.update` call in the patch will store the address twice. I assume this happens due to the calls being wrapped in a transaction, but I haven't figured out a better way yet. On our local installation we remove this line and add a before_transition to: :complete hook to store the defaults on the user.

Reply:
@mrbrdo you found a solution this in https://github.com/spree/spree/pull/11660 

```
user.class.unscoped.where(id: user).update_all(bill_address_id: bill_address.id) if user && user.bill_address.nil?
```